mozilla-services/merino-py
# [DISCO-2806]: Add location completion via accuweather api

## References

JIRA: [DISCO-2806](https://mozilla-hub.atlassian.net/browse/DISCO-2806)

## Description
Adding functionality to provide location completion suggestions via Accuweather api by modifying our existing `/suggest` endpoint.

## Testing
- Make sure your `default.local.toml` has the correct settings and the `accuweather_city` provider's `cache` is set to `redis`. 
- Start the redis instance: `cd dev && docker-compose up`
- Start merino `make dev`

#### Backwards compatibility tests
This should return empty suggestions:
`curl -i -X 'GET' 'localhost:8000/api/v1/suggest?q=new&providers=accuweather&request_type=location'`

This should return weather suggestions:
`curl -i -X 'GET' 'localhost:8000/api/v1/suggest?q=&providers=accuweather&request_type=weather'`

This should also return weather suggestions:
`curl -i -X 'GET' 'localhost:8000/api/v1/suggest?q=&providers=accuweather'`

#### New provider tests
This should return location completion suggestions:
`curl -i -X 'GET' 'localhost:8000/api/v1/suggest?q=new&providers=accuweather_city&request_type=location`

This should return weather suggestions:
`curl -i -X 'GET' 'localhost:8000/api/v1/suggest?q=new&providers=accuweather_city&request_type=weather'`

This should return an error (400):
`curl -i -X 'GET' 'localhost:8000/api/v1/suggest?q=new&providers=accuweather_city&request_type=unsupported'`

## PR Review Checklist

_Put an `x` in the boxes that apply_

- [x] The PR title starts with the JIRA issue reference, format `[DISCO-####]`, and has the same title (if applicable)
- [ ] `[load test: (abort|warn)]` keywords are applied (if applicable)


[DISCO-2806]: https://mozilla-hub.atlassian.net/browse/DISCO-2806?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
